listenbrainz/webserver/static/js/src/RecentListens.tsx

```diff
@@ -660,6 +660,18 @@ export default class RecentListens extends React.Component<
                       return 0;
                     })
                     .map((listen) => {
+                      const isCurrentUser =
```

Comment:
Similar checks are present in deleteListen method to not call the API method when clicking delete listen on another user's profile. Not sure if necessary to have the same check in two places. I am leaning in favour of just having the checks here and assume in deleteListen that if the method has been called its valid to delete the listen. Thoughts?

Reply:
I don't mind the belt-and-braces approach, considering they are two separate concerns (one to restrict display elements, one to restrict business logic).

My only real argument is that it could make unit testing more difficult, i.e. we couldn't test calling the deleteListen function in a test without a logged in user to ensure it doesn't call the API.
I suppose with react devtools I could manually call that method, bypassing the need for a button to be displayed, but after all all they'll get is probably an error of some sort.

Reply:
Yup, with devtools are bets are off anyways. I see testing is indeed easier this way. Cool, let's leave it as is then.